api/tests/Api/SnapshotTests/ResponseSnapshotTest.php

```diff
@@ -34,20 +39,28 @@ public function testRootEndpointMatchesSnapshot() {
     }
 
     public function testOpenApiSpecMatchesSnapshot() {
-        $response = static::createClientWithCredentials()
-            ->request(
-                'GET',
-                '/docs.json',
-                [
-                    'headers' => [
-                        'accept' => 'application/json',
-                    ],
-                ]
-            )
-        ;
+        $container = static::$kernel->getContainer();
+
+        /** @var OpenApiFactoryInterface $openApiFactory */
+        $openApiFactory = $container->get('api_platform.openapi.factory');
+
+        /** @var NormalizerInterface $normalizerInterface */
+        $normalizerInterface = $container->get('api_platform.openapi.normalizer.api_gateway');
+
+        $openApiResult = $openApiFactory->__invoke();
+        $openApiResultArray = $normalizerInterface->normalize($openApiResult, 'json');
+        $sortedOpenApiArray = ArrayDeepSort::sort($openApiResultArray);
```

Comment:
Instead of doing this whole dance, why don't you just keep on using `$response->toArray()` and then sort and convert that? This is an ECampApiTestCase, you are allowed to perform a request.

Reply:
The endpoint seems to be not available anymore in api platform 3.2, or does not support
json at least. (or i did not find it).

The api-platform docs also reference the cli command to generate the docs.

Reply:
What do you mean, the endpoint seems to be not available anymore? Before your changes, this test ran by requesting this endpoint, so it can't have disappeared.
Maybe we misunderstand each other? I am proposing something as follows:
```php
        $response = static::createClientWithCredentials()
            ->request(
                'GET',
                '/docs.json',
                [
                    'headers' => [
                        'accept' => 'application/json',
                    ],
                ]
            )
        ;

        $this->assertResponseStatusCodeSame(200);
        $responseArray = $response->toArray(false);
        $sortedResponseArray = ArrayDeepSort::sort($responseArray);
        $this->assertMatchesSnapshot($sortedResponseArray, new ECampYamlSnapshotDriver());
```
This is way closer to the previous JSON snapshot implementation. We could even clean this up more, by modifying our own `assertMatchesResponseSnapshot` function to accept a format or driver argument:
```php
        $response = static::createClientWithCredentials()
            ->request(
                'GET',
                '/docs.json',
                [
                    'headers' => [
                        'accept' => 'application/json',
                    ],
                ]
            )
        ;

        $this->assertResponseStatusCodeSame(200);
        $this->assertMatchesResponseSnapshot($response, new ECampYamlSnapshotDriver());
```

Reply:
in api platform 3.2 they were not available anymore

Reply:
Well, it must be available somewhere as JSON or even YAML, otherwise the Swagger UI couldn't work, and I am quite sure that API platform wouldn't just abandon Swagger.

Anyways, could we focus on just switching to YAML snapshots in this PR and tackle the api platform 3.2 changes when we upgrade to that version? Or alternatively, you could mark this PR as draft and we can wait until we have upgraded to 3.2.